core/state/access_list.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2020 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package state
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type accessList struct {
+	addresses map[common.Address]int
+	slots     []map[common.Hash]bool
+}
+
+// ContainsSlot returns true if the address is in the access list
+func (al *accessList) ContainsAddr(address common.Address) bool {
+	_, ok := al.addresses[address]
+	return ok
+}
+
+// ContainsSlot returns whether the (addres, slot) is present.
+func (al *accessList) ContainsSlot(address common.Address, slot common.Hash) (addressPresent bool, slotPresent bool) {
```

Comment:
I'd rename to Contains. Since we don't have a method to check an address, only a slot, seems like a simpler API.